File: Lab14/Lab14_2/app/src/main/java/com/example/costumlistview/MyArrayAdapter.java
package com.example.costumlistview;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageButton;
import android.widget.TextView;

import java.util.ArrayList;

public class MyArrayAdapter extends ArrayAdapter<Item> {
    Activity context;
    int layoutId;
    ArrayList<Item> list;

    public MyArrayAdapter(Activity context, int layoutId, ArrayList<Item> list) {
        super(context, layoutId, list);
        this.context = context;
        this.layoutId = layoutId;
        this.list = list;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = context.getLayoutInflater();
        convertView = inflater.inflate(layoutId, null);

        TextView txtMaso = convertView.findViewById(R.id.txtmaso);
        TextView txtTieude = convertView.findViewById(R.id.txttieude);
        ImageButton btnLike = convertView.findViewById(R.id.btnlike);

        Item item = list.get(position);
        txtMaso.setText(item.getMaso());
        txtTieude.setText(item.getTieude());

        if (item.getThich() == 1) {
            btnLike.setImageResource(R.drawable.liked);
        } else {
            btnLike.setImageResource(R.drawable.unlike);
        }

        return convertView;
    }
}

File: Lab24/app/src/main/java/com/example/parsejson_bank/MyArrayAdapter.kt
package com.example.parsejson_bank

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView

class MyArrayAdapter(
    private val contextActivity: Activity,
    private val resourceLayout: Int,
    private val objectsList: List<Tygia>
) : ArrayAdapter<Tygia>(contextActivity, resourceLayout, objectsList) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val inflater: LayoutInflater = contextActivity.layoutInflater
        val itemView = inflater.inflate(resourceLayout, null)

        val tygia = objectsList[position]

        val imgHinh: ImageView = itemView.findViewById(R.id.imgHinh)
        val txtType: TextView = itemView.findViewById(R.id.txtType)
        val txtMuaTM: TextView = itemView.findViewById(R.id.txtMuaTM)
        val txtBanTM: TextView = itemView.findViewById(R.id.txtBanTM)
        val txtMuaCK: TextView = itemView.findViewById(R.id.txtMuaCK)
        val txtBanCK: TextView = itemView.findViewById(R.id.txtBanCK)

        imgHinh.setImageBitmap(tygia.bitmap)
        txtType.text = tygia.type
        txtMuaTM.text = tygia.muatienmat
        txtBanTM.text = tygia.bantuenmat
        txtMuaCK.text = tygia.muack
        txtBanCK.text = tygia.banck

        return itemView
    }
}
